plugins/discourse-lazy-videos/lib/lazy-videos/lazy_vimeo.rb

```diff
@@ -8,16 +8,25 @@ class Onebox::Engine::VimeoOnebox
 
   def to_html
     if SiteSetting.lazy_videos_enabled && SiteSetting.lazy_vimeo_enabled
-      video_id = oembed_data[:video_id]
+      full_video_id = oembed_data[:uri].sub("/videos/", "").sub(":", "/")
+
+      if !oembed_data[:uri].match?(%r{videos/\d+:.+})
+        iframe_id = full_video_id
+      else
+        iframe_src = Nokogiri::HTML5.fragment(oembed_data[:html]).at_css("iframe")&.[]("src")
+        iframe_id = iframe_src.sub("https://player.vimeo.com/video/", "")
+      end
+
       thumbnail_url = "https://vumbnail.com/#{oembed_data[:video_id]}.jpg"
+      puts thumbnail_url
       escaped_title = ERB::Util.html_escape(og_data.title)
 
       <<~HTML
         <div class="vimeo-onebox lazy-video-container"
-          data-video-id="#{video_id}"
+          data-video-id="#{iframe_id}"
           data-video-title="#{escaped_title}"
           data-provider-name="vimeo">
-          <a href="https://vimeo.com/#{video_id}" target="_blank">
+          <a href="https://vimeo.com/#{full_video_id}" target="_blank">
```

Comment:
Very minor, but both `iframe_id` and `full_video_id` don't seem quite right. Do we need to differentiate them?

Reply:
It got confusing, but it's not entirely my fault. Essentially, there are three IDs. 
Let's use this one as an example: https://vimeo.com/192207770/0faf1dd09d

**oembed_data[:video_id]:** `192207770` (used just for the image, even on unlisted videos)
**full_video_id:**  `192207770/0faf1dd09d` (used for the link, it can be the same as the oembed_data id)
**iframe_id:** `192207770?h=0faf1dd09d&amp;app_id=122963` (used for the iframe, taken from the iframe src)

Reply:
Yeah, it's tricky, definitely not your fault, naming in general is hard and the edge case here doesn't make this easy. Let's go with this for now. 

(Later on, it might be good to refactor this and pass a fully URL as the `iframeSrc` to the frontend code, might be easier to reason about and test.) 

Thanks, approving!